### PR TITLE
fix: token retrieval error handling

### PR DESCRIPTION
**Pull request by attriaayush**
Signed-off-by: Aayush Attri <aayush.attri@snyk.io>

### Description

The token retrieval method within `SecretStorage` class throws an error when the token cannot be parsed which creates a generic error that we want to handle for.

Also removing a `console.log` within `src/snyk/common/services/learnService.ts`

### Checklist

- [ ] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

**Review by Geit — DISMISSED** [review]
LGTM (and apologies!) for learn

File: src/snyk/common/configuration/configuration.ts [hunk 1]
```diff
@@ -167,7 +167,12 @@ export class Configuration implements IConfiguration {
   }
 
   async getToken(): Promise<string | undefined> {
-    return SecretStorageAdapter.instance.get(SNYK_TOKEN_KEY);
+    try {
+      return SecretStorageAdapter.instance.get(SNYK_TOKEN_KEY);
+    } catch (e) {
+      // if the token cannot be parsed then clear the token
```
Comment (michelkaporin) [inline src/snyk/common/configuration/configuration.ts:173]: Should we additionally log the error to output channel? It's still a mystery when this fails by default.

**Review by michelkaporin — COMMENTED** [review]
Probably worth adding to changelog since it's user-facing 😉

File: src/snyk/common/configuration/configuration.ts (hunk 1 above)
Comment (attriaayush) [reply src/snyk/common/configuration/configuration.ts:173]: Yeah I thought about adding a logger here but then the only error we will be catching is the [`Cannot get password` ](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/api/browser/mainThreadSecretState.ts#L48) which isn't very useful to start with.

**Review by michelkaporin — DISMISSED** [review]

**Review by michelkaporin — APPROVED** [review]

**Pull request merged**